internal/inventory/tagprocessor.go

```diff
@@ -0,0 +1,364 @@
+		}
+
+		locationMean := statsAtPrevLoc.rssiDbm.Mean()
+		incomingMean := statsAtReadLoc.rssiDbm.Mean()
+
+		offset := tp.config.profile.computeOffset(info.referenceTimestamp, statsAtPrevLoc.LastRead)
+		if tp.config.debugLogEnabled {
+			logTagStats(tp, tag, readLocation.String(), incomingMean, locationMean, offset)
+		}
+
+		// Update the location if the mean RSSI at the new location
+		// is greater than the adjusted mean RSSI of the existing location.
+		// Note: This will generate a moved event.
+		if incomingMean > (locationMean + offset) {
+			tag.Location = readLocation
+		}
+	}
+
+	return
+}
+
+func logTagStats(tp *TagProcessor, tag *Tag, readLocation string, incomingMean float64, existingMean float64, offset float64) {
+	tp.lc.Debug("tag stats",
+		"epc", tag.EPC,
+		"readLoc", readLocation,
+		"prevLoc", tag.Location,
+		"incomingAvg", fmt.Sprintf("%.2f", incomingMean),
+		"existingAvg", fmt.Sprintf("%.2f", existingMean),
+		"offset", fmt.Sprintf("%.2f", offset),
+		"existingAdjusted", fmt.Sprintf("%.2f", existingMean+offset),
+		// if stayFactor is positive, tag will stay, if negative, generates a moved event
+		"stayFactor", fmt.Sprintf("%.2f", (existingMean+offset)-incomingMean))
+}
+
+func logReadTiming(tp *TagProcessor, info ReportInfo, locationStats *TagStats, tag *Tag) {
+	now := UnixMilliNow()
+	tp.lc.Debug("read timing",
+		"now", now,
+		"referenceTimestamp", info.referenceTimestamp,
+		"nowMinusRef", fmt.Sprintf("%v", time.Duration(now-info.referenceTimestamp)*time.Millisecond),
+		"locationLastRead", locationStats.LastRead,
+		"lastRead", tag.LastRead,
+		"diff", fmt.Sprintf("%v", time.Duration(tag.LastRead-locationStats.LastRead)*time.Millisecond))
+}
+
+// AgeOut is a cleanup method that will remove tag information from our in-memory
+// structures if it has not been seen in a long enough time. Only applies to
+// tags which are already Departed.
+func (tp *TagProcessor) AgeOut() (int, []StaticTag) {
+	expiration := UnixMilli(time.Now().Add(time.Hour * time.Duration(-tp.config.ageOutHours)))
+
+	// developer note: Go allows us to remove from a map while iterating
+	var numRemoved int
+	for epc, tag := range tp.inventory {
+		if tag.state == Departed && tag.LastRead < expiration {
+			numRemoved++
+			delete(tp.inventory, epc)
+		}
+	}
+
+	if numRemoved > 0 {
+		tp.lc.Info(fmt.Sprintf("Inventory ageout removed %d tag(s).", numRemoved))
+		return numRemoved, tp.snapshot()
+	}
+
+	tp.lc.Debug("No tags were aged-out.")
+	return 0, nil
+}
+
+// AggregateDeparted loops through all tags and sees if any of them should be Departed
+// due to not being read in a long enough time.
+func (tp *TagProcessor) AggregateDeparted() (events []Event, snapshot []StaticTag) {
+	now := time.Now()
+	nowMs := now.UnixNano() / 1e6
+	expiration := now.Add(-time.Duration(tp.config.departedThresholdSeconds)*time.Second).UnixNano() / 1e6
+
+	for _, tag := range tp.inventory {
+		if tag.state == Present && tag.LastRead < expiration {
```

Comment:
Same issue here with `< expiration`

Reply:
Same as above. The math is valid, but the choice of words is confusing. I will document and change the variable name.